packages/flutter/lib/src/material/list_item.dart

```diff
@@ -5,53 +5,123 @@
 import 'package:flutter/widgets.dart';
 
 import 'ink_well.dart';
+import 'theme.dart';
 
+/// Material List items are one to three lines of text optionally flanked by icons.
+/// Icons are defined with the [left] and [right] parameters. The first line of text
+/// is not optional and is specified with [primary]. The value of [secondary] will
+/// occupy the space allocated for an aditional line of text, or two lines if
+/// isThreeLine: true is specified. If isDense: true is specified then the overall
+/// height of this list item and the size of the DefaultTextStyles that wrap
+/// the [primary] and [secondary] widget are reduced.
 class ListItem extends StatelessComponent {
   ListItem({
     Key key,
     this.left,
```

Comment:
We might want to rename these left/right properties because they'll be reversed in RTL layouts.  Maybe "avatar" and "action" ?